File: backend/src/main/java/com/daedan/festabook/lineup/dto/LineupResponse.java
package com.daedan.festabook.lineup.dto;

import com.daedan.festabook.lineup.domain.Lineup;
import java.time.LocalDateTime;

public record LineupResponse(
        Long lineupId,
        String name,
        String imageUrl,
        LocalDateTime performanceAt
) {

    public static LineupResponse from(Lineup lineup) {
        return new LineupResponse(
                lineup.getId(),
                lineup.getName(),
                lineup.getImageUrl(),
                lineup.getPerformanceAt()
        );
    }
}

File: android/app/src/main/java/com/daedan/festabook/data/model/response/lineup/LineupResponse.kt
package com.daedan.festabook.data.model.response.lineup

import com.daedan.festabook.domain.model.LineupItem
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.LocalDateTime

@Serializable
data class LineupResponse(
    @SerialName("lineupId")
    val lineupId: Long,
    @SerialName("imageUrl")
    val imageUrl: String,
    @SerialName("name")
    val name: String,
    @SerialName("performanceAt")
    val performanceAt: String,
)

fun LineupResponse.toDomain(): LineupItem =
    LineupItem(
        id = lineupId,
        imageUrl = imageUrl,
        name = name,
        performanceAt = LocalDateTime.parse(performanceAt),
    )
